### PR TITLE
Associate person, book, rental and classroom

### PR DESCRIPTION
**Pull request by Haadiiii**
## OOP School library Association class Project :arrow_double_down: 

In this Activity : 

>Classroom.rb

- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).

>Association Between Students and Classrooms.

- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

>Book.rb

- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).

>Rental.rb

- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).

>Association between Rentals, Books and persons

- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author
```
Comment (FrederickMih) [inline book.rb:2]: - [ ] For a book to have many rentals, Getters and Setters for @rentals should be present.

`attr_accessor :title, :author, :rentals` . Let's kindly fix :wrench: it.

File: person.rb [hunk 2]
```diff
@@ -4,16 +4,22 @@
 require './trimmerdecorator'
 
 class Person < Nameable
+  attr_reader :id, :rental
+  attr_accessor :name, :age
```
Comment (FrederickMih) [inline person.rb:8]: - [ ] A person has many rentals hence Getters and Setters for @rentals should be present:  

`attr_accessor :name, :age, :rentals` Kindly fix :wrench: it

File: person.rb [hunk 3]
```diff
@@ -4,16 +4,22 @@
...
+
   def initialize(age, name = 'unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
```
Comment (FrederickMih) [inline person.rb:16]: :+1:  @rentals. Let's pluralize it. So many rentals :smile:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Haadiiii  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 

- Correct Gitflow has been followed :heavy_check_mark: 
- Pull Request has a title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 


## Required Changes ♻️

- [ ] Please customized all the content in your README.md file to match your project. Your README.md file should be descriptive and professional. 



_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Haadiiii) [reply book.rb:2]: got it :+1:

File: person.rb (hunk 2 above)
Comment (Haadiiii) [reply person.rb:8]: :smile:  :+1:

File: person.rb (hunk 3 above)
Comment (Haadiiii) [reply person.rb:16]: @FrederickMih  :+1: :smile:

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢🟢🟢 

Hi @Haadiiii,

Good job on making the changes asked by the previous reviewer.

Your project is complete!  

## Highlights:
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ Good descriptive PR 👍 

There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**